### PR TITLE
[PF-511] Logging package: fix unit test and spotbugs errors

### PR DESCRIPTION
**Pull request by gjuggler**
Fixes a couple bugs in the verification / static checks for the logging package.

I hadn't fully tested the `LoggingUtils` init method before merging #19. Both unit test classes work fine when run independently, but either one fails (whichever one is run second) when run together as part of a single JUnit execution.

The problem was with the `loggingInitialized` boolean. It works fine when only one app initialization flow is occurring within the lifetime of the Java execution, but in the test environment a new app context is initialized for each test class.

The cleanest fix was to add some code which explicitly resets the Logback context and reinstates human-readable logging, to complement the existing code which resets Logback and reinstates JSON logging.

This PR also contains a drive-by cleanup, to make the logging tests use the newer JUnit Jupiter test APIs rather than legacy JUnit 4 equivalents.

### CODE REVIEW

**Review by wchamber — APPROVED** [review]

File: src/main/java/bio/terra/common/logging/LoggingUtils.java [hunk 1]
```diff
@@ -78,34 +77,30 @@ static JsonNode jsonFromString(String s) throws JsonProcessingException {
...
-
     ch.qos.logback.classic.Logger logbackLogger =
         (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
     ConfigurableEnvironment environment = applicationContext.getEnvironment();
 
     if (Arrays.stream(environment.getActiveProfiles()).anyMatch("human-readable-logging"::equals)) {
       System.out.println("Human-readable logging enabled, skipping Google JSON layout");
-      return;
-    }
+      // No action needed. The logback.xml file on the classpath contains a default human-readable
```
Comment (jaycarlton) [inline src/main/java/bio/terra/common/logging/LoggingUtils.java:86]: curious: If this branch is effectively the default behavior, why was it the one to get the keyword?

File: src/main/java/bio/terra/common/logging/LoggingUtils.java [hunk 2]
```diff
@@ -78,34 +77,30 @@ static JsonNode jsonFromString(String s) throws JsonProcessingException {
...
+      encoder.start();
 
-    LayoutWrappingEncoder encoder = new LayoutWrappingEncoder();
-    encoder.setLayout(layout);
-    encoder.start();
+      ConsoleAppender appender = new ConsoleAppender();
+      appender.setName(TERRA_APPENDER_NAME);
+      appender.setEncoder(encoder);
+      appender.setContext(logbackLogger.getLoggerContext());
+      appender.start();
```
Comment (jaycarlton) [inline src/main/java/bio/terra/common/logging/LoggingUtils.java:100]: 🎵 the ApplicationContext on the layout and the layout on the encoder and the encoder on the appender and the appender on the logger and the green grass grows all around all around...

File: src/test/java/bio/terra/common/logging/LoggingTest.java [hunk 3]
```diff
@@ -81,8 +77,7 @@
     @Override
     public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
         throws IOException, ServletException {
-      Scope scope =
-          Tracing.getTracer().spanBuilderWithExplicitParent("test-span", null).startScopedSpan();
+      Tracing.getTracer().spanBuilderWithExplicitParent("test-span", null).startScopedSpan();
       requestSpanContext = Tracing.getTracer().getCurrentSpan().getContext();
       chain.doFilter(request, response);
       // Note: we purposefully don't close the scope here. For some reason, it doesn't seem possible
```
Comment (jaycarlton) [inline src/test/java/bio/terra/common/logging/LoggingTest.java:83]: This sounds odd. Possibly out of scope.

File: src/test/java/bio/terra/common/logging/LoggingTest.java [hunk 4]
```diff
@@ -201,8 +196,8 @@ public void testStructuredLogging() {
             .read(path);
   }
 
-  private String lastLoggedLine() {
-    String[] lines = outputCapture.getAll().split("\n");
+  private String lastLoggedLine(CapturedOutput capturedOutput) {
```
Comment (jaycarlton) [inline src/test/java/bio/terra/common/logging/LoggingTest.java:199]: should probably be Optional<String> in case there are no lines in the capture.

**Review by jaycarlton — APPROVED** [review]

File: src/test/java/bio/terra/common/logging/LoggingTest.java (hunk 4 above)
Comment (gjuggler) [reply src/test/java/bio/terra/common/logging/LoggingTest.java:199]: That might be a tad cleaner, but IMO it's good enough for the test to fail on an NPE here. If you're asking for the last logged line and none exists, the case should fail hard & fast.

File: src/test/java/bio/terra/common/logging/LoggingTest.java (hunk 3 above)
Comment (gjuggler) [reply src/test/java/bio/terra/common/logging/LoggingTest.java:83]: :) it's a condensation of a much more frustrating debugging session than one might imagine...

**Pull request merged**